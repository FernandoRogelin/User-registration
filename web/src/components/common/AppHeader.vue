<script setup>
const props = defineProps({
  step: {
    type: Number,
    required: true
  },
  isCompany: {
    type: Boolean,
    default: false
  }
})

function getStepTitle() {
  const steps = {
    1: 'Seja bem-vindo(a)',
    2: `Pessoa ${props.isCompany ? 'Jurídica' : 'Física'}`,
    3: 'Senha de acesso',
    4: 'Revise suas informações'
  }

  return steps[props.step] || '';
}
</script>

<template>
  <header class="components-common-appHeader">
    <p class="components-common-appHeader-step">
      Etapa <span class="components-common-appHeader-step__number">{{ step }}</span> de 4
    </p>

    <h1 class="components-common-appHeader-title">{{ getStepTitle() }}</h1>
  </header>
</template>

<style scoped lang="scss">
.components-common-appHeader{
  &-step {
    font-size: 0.8rem;
    margin-bottom: 0.4rem;

    &__number {
      color: var(--mb-orange);
    }
  }

  &-title {
    font-size: 1.4rem;
  }
}
</style>
