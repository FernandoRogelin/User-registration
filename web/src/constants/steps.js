export const FIRST_STEP = 1
export const SECOND_STEP = 2
export const THIRD_STEP = 3
export const LAST_STEP = 4

export const stepFields = {
  1: ['email', 'personType'],
  2: {
    person: ['name', 'cpf', 'birthdate', 'phone'],
    company: ['companyName', 'cnpj', 'openingDate', 'companyPhone']
  },
  3: ['password']
}
