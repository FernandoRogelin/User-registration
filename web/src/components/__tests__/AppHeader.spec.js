import { describe, expect, test, afterEach } from "vitest"
import { render, screen, cleanup } from "@testing-library/vue"

import AppHeader from "@/components/common/AppHeader.vue"

import { FIRST_STEP, LAST_STEP, SECOND_STEP, THIRD_STEP } from "@/constants/steps"

afterEach(() => cleanup())

describe('<AppHeader /> common component', () => {
  test('should render the component and match snapshot', () => {
    const { container } = render(AppHeader, { props: { step: FIRST_STEP } })

    expect(container.firstChild).toMatchSnapshot()
  })

  test('should render the first step', () => {
    render(AppHeader, { props: { step: FIRST_STEP } })

    expect(screen.getByRole('heading', { name: /Seja bem-vindo\(a\)/i })).toBeInTheDocument()
  })

  test('should render the second step private individuals', () => {
    render(AppHeader, { props: { step: SECOND_STEP, isCompany: false } })

    expect(screen.getByRole('heading', { name: 'Pessoa Física' })).toBeInTheDocument()
  })

  test('should render the second step legal person', () => {
    render(AppHeader, { props: { step: SECOND_STEP, isCompany: true } })

    expect(screen.getByRole('heading', { name: 'Pessoa Jurídica' })).toBeInTheDocument()
  })

  test('should render the third step', () => {
    render(AppHeader, { props: { step: THIRD_STEP } })

    expect(screen.getByRole('heading', { name: /Senha de acesso/i })).toBeInTheDocument()
  })

  test('should render the last step', () => {
    render(AppHeader, { props: { step: LAST_STEP } })

    expect(screen.getByRole('heading', { name: /Revise suas informações/i })).toBeInTheDocument()
  })
})
