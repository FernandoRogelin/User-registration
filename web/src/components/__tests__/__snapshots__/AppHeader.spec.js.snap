// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<AppHeader /> common component > should render the component and match snapshot 1`] = `
<header
  class="components-common-appHeader"
  data-v-1820f6ff=""
>
  <p
    class="components-common-appHeader-step"
    data-v-1820f6ff=""
  >
     Etapa 
    <span
      class="components-common-appHeader-step__number"
      data-v-1820f6ff=""
    />
     de 4 
  </p>
  <h1
    class="components-common-appHeader-title"
    data-v-1820f6ff=""
  />
</header>
`;
